app/modules/ph/endpoints_ph.py

```diff
@@ -0,0 +1,231 @@
+        filename=str(paper_id),
+    )
+
+
+@module.router.get(
+    "/ph/",
+    response_model=list[schemas_ph.PaperComplete],
+    status_code=200,
+)
+async def get_papers(
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user_a_member),
+):
+    """
+    Return all editions, sorted from the latest to the oldest
+    """
+    result = await cruds_ph.get_papers(
+        db=db,
+    )  # Return papers from the latest to the oldest
+    return result
+
+
+@module.router.post(
+    "/ph/",
+    response_model=schemas_ph.PaperComplete,
+    status_code=201,
+)
+async def create_paper(
+    paper: schemas_ph.PaperBase,
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user_a_member_of(GroupType.ph)),
+    notification_tool: NotificationTool = Depends(get_notification_tool),
+):
+    """Create a new paper."""
+
+    paper_complete = schemas_ph.PaperComplete(
+        id=str(uuid.uuid4()),
+        **paper.model_dump(),
+    )
+    try:
+        paper_db = models_ph.Paper(
+            id=paper_complete.id,
+            name=paper_complete.name,
+            release_date=paper_complete.release_date,
+        )
+
+        now = datetime.now(UTC)
+        message = Message(
+            context=f"ph-{paper_db.id}",
+            is_visible=True,
+            title=f"📗 PH - {paper_db.name}",
+            content="A new paper has been released! 🎉",
+            # The notification will expire in 3 days
+            expire_on=now.replace(day=now.day + 3),
+        )
+        await notification_tool.send_notification_to_topic(
+            custom_topic=CustomTopic(topic=Topic.advert),
+            message=message,
+        )
+        return await cruds_ph.create_paper(paper=paper_db, db=db)
+
+    except ValueError as error:
+        raise HTTPException(
+            status_code=400,
+            detail=str(error),
+        )
+
+
+@module.router.post(
+    "/ph/{paper_id}/pdf",
+    response_model=standard_responses.Result,
+    status_code=201,
+)
+async def create_paper_pdf(
+    paper_id: str,
+    pdf: UploadFile = File(...),
+    user: models_core.CoreUser = Depends(is_user_a_member_of(GroupType.ph)),
+    request_id: str = Depends(get_request_id),
+    db: AsyncSession = Depends(get_db),
+):
+    paper = await cruds_ph.get_paper_by_id(db=db, paper_id=paper_id)
+    if paper is None:
+        raise HTTPException(
+            status_code=404,
+            detail="The paper does not exist.",
+        )
+
+    await save_file_as_data(
+        image=pdf,
+        directory="ph/pdf",
+        filename=str(paper_id),
+        request_id=request_id,
+        max_file_size=10 * 1024 * 1024,
+        accepted_content_types=["application/pdf"],
+    )
+    Path("data/ph/cover/").mkdir(parents=True, exist_ok=True)
+    paper_pdf = fitz.open(f"data/ph/pdf/{paper_id}.pdf")
+    page = paper_pdf.load_page(0)
+    pix = page.get_pixmap(matrix=fitz.Matrix(300 / 72, 300 / 72))
```

Comment:
Could you add a comment explaining how does the matrix multiplication works, and why the value `300/72` was chosen?

Reply:
Are you sure it's useful to increase the image size by 4? It seams pretty big for a cover

Reply:
I just took a basic example on the internet without understand it for the tests but now, i modified it and it's better.